analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+            => new AnalysisContext(explodedGraph);
+
+        private sealed class AnalysisContext : ISymbolicExecutionAnalysisContext
+        {
+            private readonly List<Location> locations = new List<Location>();
+
+            public bool SupportsPartialResults => false;
+
+            public AnalysisContext(AbstractExplodedGraph explodedGraph) =>
+                explodedGraph.AddExplodedGraphCheck(new InitializationVectorCheck(explodedGraph, this));
+
+            public IEnumerable<Diagnostic> GetDiagnostics() =>
+                locations.Distinct().Select(location => Diagnostic.Create(Rule, location));
+
+            public void AddLocation(Location location) => locations.Add(location);
+
+            public void Dispose()
+            {
+                // Nothing to do here.
+            }
+        }
+
+        private sealed class InitializationVectorCheck : ExplodedGraphCheck
+        {
+            private static readonly ImmutableArray<KnownType> VulnerableTypes =
+                ImmutableArray.Create(KnownType.System_Security_Cryptography_Aes,
+                                      KnownType.System_Security_Cryptography_AesCng,
+                                      KnownType.System_Security_Cryptography_AesCryptoServiceProvider,
+                                      KnownType.System_Security_Cryptography_SymmetricAlgorithm);
+
+            private readonly AnalysisContext context;
+
+            public InitializationVectorCheck(AbstractExplodedGraph explodedGraph, AnalysisContext context) : base(explodedGraph) =>
+                this.context = context;
+
+            public override ProgramState PostProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                programState = programPoint.CurrentInstruction switch
+                {
+                    ObjectCreationExpressionSyntax objectCreation => ObjectCreationPostProcess(objectCreation, programState),
+                    InvocationExpressionSyntax invocation => InvocationExpressionPostProcess(invocation, programState),
+                    ArrayCreationExpressionSyntax arrayCreation => ArrayCreationPostProcess(arrayCreation, programState),
+                    AssignmentExpressionSyntax assignment => AssignmentExpressionPostProcess(assignment, programState),
+                    _ => programState
+                };
+
+                return base.PostProcessInstruction(programPoint, programState);
+            }
+
+            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                if (programPoint.CurrentInstruction is InvocationExpressionSyntax invocation)
+                {
+                    programState = InvocationExpressionPreProcess(invocation, programState);
+                }
+
+                return base.PreProcessInstruction(programPoint, programState);
+            }
```

Comment:
Similar to previous method, can be shortened
```suggestion
            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState) =>
                programPoint.CurrentInstruction is InvocationExpressionSyntax invocation
                    ? InvocationExpressionPreProcess(invocation, programState)
                    : programState;

```